Fitter/src/net/relinc/fitter/GUI/HomeController.java

```diff
@@ -209,6 +209,34 @@ public void handle(MouseEvent event) {
 		
 		
 		
+	}
+
+	public void setFitterFromTrimSelection(Double xAxisBegin, Double xAxisEnd){
+		FitableDataset theData = datasetsListView.getSelectionModel().getSelectedItem();
+		theData.setBeginFromXValue(xAxisBegin);
+		theData.setEndFromXValue(xAxisEnd);
+
+		FitableDataset f = getCurrentDataset();
+		if(f == null)
+			return;
```

Comment:
no need to call `renderCharts()` before returning here? Before your last commit, it would have called `renderCharts()`.

Reply:
There is no need to call renderCharts() before returning. In TrimDataController.applyModifierButtonFired() c.renderCharts() is called right before c.setFitterFromTrimSelection() (new method). If you comment out the new method the old rendering will occur. If you change if( f == null) to if(true) in this case the old rendering occurs.

Reply:
when you change if( f==null) to if(true) there is a side effect of the new rendering occuring when you click on zoom rectangle or when you adjust the order of polynomial fit in which the new fitting occurs. But that only happens because if(f == null) is false in their respective methods.